services/src/main/java/org/fao/geonet/api/records/formatters/XsltFormatter.java

```diff
@@ -97,16 +115,33 @@ public String format(FormatterParams fparams) throws Exception {
 
         root.addContent(new Element("lang").setText(fparams.context.getLanguage()));
         root.addContent(new Element("url").setText(fparams.url));
-        // FIXME: This is a hack to mimic what Jeeves service are doing.
-        // Some XSLT are used by both formatters and Jeeves and Spring MVC services
-        Element translations = new Element("translations");
+
+        if (!translationElements.containsKey(fparams.context.getLanguage())) {
+            // Get translation keys and add them to the cache
+            Element translations = new Element("translations");
+            Map<String, String> translationMap = new JSONLocCacheLoader(configurableApplicationContext, fparams.context.getLanguage()).call();
+            for (Map.Entry<String, String> entry : translationMap.entrySet()) {
+                // Skip keys that are not alpha numeric only including "." - otherwhise chars like : ? +... cause problem when creating element.
+                if (entry.getKey().matches("[a-zA-Z0-9\\.]+")) {
+                    try {
+                        translations.addContent(new Element(entry.getKey()).setText(entry.getValue()));
+                    } catch (Exception e) {
+                        System.out.println("Failed to add translation key for \"" + entry.getKey() + "\"=\"" + entry.getValue() + "\". " + e.getMessage());
```

Comment:
This should be changed to log the exception.

Reply:
Changed to log the exception